src/javaharness/java/arcs/crdt/CRDTCollection.java

```diff
@@ -0,0 +1,174 @@
+package arcs.crdt;
+
+import java.lang.Math;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+// Classes and interfaces copied from src/runtime/crdt/crdt-collection.ts
+interface Referenceable {
+  String getId();
```

Comment:
Feel like we should at least briefly document expectations of ID -- can it be blank, should it be opaque to CRDT internals or is there an expected format? I realize it isn't documented in the original you're porting. Maybe we should document it there, and then the header comment you have above would indirectly be referencing.

Reply:
@shans @mykmartin could you, maybe add some documentation in the ts file?